### ISSUE TITLE
CompactStorage v2.0.0.8 - Client Crash if Chest Builder's colorField has a non-alphanumeric character

### ISSUE THREAD

**Issue by mallrat208** [tattyseal/IssueTracker#5]
CompactStorage version 2.0.0.8
Crash Report: https://gist.github.com/mallrat208/ff7b9826a71882603ac3
FML Client Latest: Way too large to paste, thanks Agrarian Skies 2 :)

This is courtesy of me fat-fingering a value while trying to change the color of a chest and holding Shift while typing 1.

```java
        if(!colorField.getText().isEmpty())
        {
            color = Integer.decode("0x" + colorField.getText());
        }
```

There's no check to verify the text field actually contains only alphanumeric characters when trying to decode the hex value. There appears to be a check to allow only certain keys, but it may be worth adding a check to make sure shift isn't actively being pressed. Or maybe validate the value against a regex expression.

**Comment by tattyseal** [comment]
This is an issue, will be fixed soon.

**Comment by tattyseal** [comment]
Will be fixed in 3.0 with the GUI overhauls. Won't be fixed in 2.0. If this becomes a bigger issue I might fix it. But it seems like a waste of time as in 3.0 I will probably have hard coded colors or a color grid to choose from so...
